### ISSUE TITLE
[Bug]: test_eagle3_qwen3_8b_sharegpt fails acceptance threshold at token 0

### ISSUE THREAD

**Issue by rahul-tuli** [vllm-project/speculators#888]
## Your current environment

- **vLLM** — `0.25.1`
- **Speculators** — `60cd983` (main, pre-release)
- **CUDA** — 13.0
- **PyTorch** — `2.12.1+cu130` test venv — **before** the `<=2.13.0` bump
- **Transformers** — `5.14.1`
- **Hardware** — 1× H100 80 GB, `ibm-wdc-k8s-h100-solo`
- **Model** — `Qwen/Qwen3-8B` (EAGLE-3, sharegpt)

## 🐛 Describe the bug

Sub-issue of #885.

`tests/e2e/regression/test_training_only_acceptance.py::test_eagle3_qwen3_8b_sharegpt` fails its acceptance assertion:

```
Acceptance 0.39957939011566773 at token 0 is less than threshold 0.4
```

Not a crash — an assertion on a trained model's quality metric. No CUDA fault, no OOM, no worker abort.

### Failing jobs

Both in run [`30138641585`](https://github.com/neuralmagic/llm-compressor-testing/actions/runs/30138641585) (2026-07-25):

| job | python | observed `acceptance_at_token_0` | threshold | shortfall |
|---|---|---|---|---|
| [`89627475848`](https://github.com/neuralmagic/llm-compressor-testing/actions/runs/30138641585/job/89627475848) | 3.10 | `0.39957939011566773` | `0.4` | **0.105 %** |
| [`89627475794`](https://github.com/neuralmagic/llm-compressor-testing/actions/runs/30138641585/job/89627475794) | 3.13 | `0.38689866939611056` | `0.4` | 3.3 % |

2 occurrences across the 4 archived runs that executed this test; both in the 07-25 run, none since.

### Not a torch 2.13 regression

Both occurrences are on torch `2.12.1+cu130`, **before** #869 raised the bound. Neither torch-`2.13.0` run reproduced it, and the test passed in every job of the v0.7.0 release run. Filed so the release-CI signal is accounted for in full, and flagged so it is not folded into the torch-2.13 work.

### Open questions

- `89627475848` missed by roughly one part in a thousand; `89627475794` by 3.3 %. These may not share one explanation — the first is consistent with threshold marginality, the second may indicate genuine variance in trained quality.
- Whether the underlying acceptance rate has drifted, or has always sat near `0.4` with the threshold set too tight. Needs the historical metric series, not pass/fail.
- The distribution of `acceptance_at_token_0` across repeated runs at fixed versions — no repeated-run data was collected.
- Whether the `[0.4, 0.1, 0.01]` thresholds were derived from a measured distribution or chosen as round numbers.
